src/databricks/sql/client.py

```diff
@@ -90,25 +86,7 @@ def __init__(
         self.port = kwargs.get("_port", 443)
         self.disable_pandas = kwargs.get("_disable_pandas", False)
 
-        authorization_header = []
-        if kwargs.get("_username") and kwargs.get("_password"):
-            auth_credentials = "{username}:{password}".format(
-                username=kwargs.get("_username"), password=kwargs.get("_password")
-            ).encode("UTF-8")
-            auth_credentials_base64 = base64.standard_b64encode(
-                auth_credentials
-            ).decode("UTF-8")
-            authorization_header = [
-                ("Authorization", "Basic {}".format(auth_credentials_base64))
-            ]
-        elif access_token:
-            authorization_header = [("Authorization", "Bearer {}".format(access_token))]
-        elif not (
-            kwargs.get("_use_cert_as_auth") and kwargs.get("_tls_client_cert_file")
-        ):
-            raise ValueError(
-                "No valid authentication settings. Please provide an access token."
-            )
+        auth_provider = get_python_sql_connector_auth_provider(server_hostname, oauth_persistence, **kwargs)
```

Comment:
Per command above, either pass all arguments explicitly, or rely on `**kwargs` for all.

Reply:
addressed. thanks.